packages/opentelemetry-resources/src/platform/node/detectors/AwsEc2Detector.ts

```diff
@@ -44,41 +54,86 @@ class AwsEc2Detector implements Detector {
    */
   async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
     try {
+      const token = await this._fetchToken();
       const {
         accountId,
         instanceId,
         instanceType,
         region,
         availabilityZone,
-      } = await this._awsMetadataAccessor();
+      } = await this._fetchIdentity(token);
+      const hostname = await this._fetchHost(token);
+
       return new Resource({
         [CLOUD_RESOURCE.PROVIDER]: 'aws',
         [CLOUD_RESOURCE.ACCOUNT_ID]: accountId,
         [CLOUD_RESOURCE.REGION]: region,
         [CLOUD_RESOURCE.ZONE]: availabilityZone,
         [HOST_RESOURCE.ID]: instanceId,
         [HOST_RESOURCE.TYPE]: instanceType,
+        [HOST_RESOURCE.NAME]: hostname,
```

Comment:
Name and hostname are the same?

Reply:
Yes, at least in Java implementation they take Name and hostname as the same: https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk_extensions/aws_v1_support/src/main/java/io/opentelemetry/sdk/extensions/trace/aws/resource/Ec2Resource.java#L192

Reply:
The main reason for this is not knowing how to populate this :) I filed an issue to ask about why there are two similar fields

https://github.com/open-telemetry/opentelemetry-specification/issues/787